File: apps/api/aqapi/core/enum/domain/value_object/enum_value_protocol.py
from abc import ABC, abstractmethod
from aqapi.core.entity.base_entity import BaseEntity


class EnumValueProtocol(ABC):
    """翻訳なしエンティティから更新可能な値オブジェクトのインターフェース"""
    
    @abstractmethod
    def set_from_entity(self, entity: BaseEntity) -> None:
        """エンティティから値を設定する
        
        :param entity: 更新に使用するエンティティ
        """
        pass

File: packages/shared/core/enum/enumValueProtocol.ts
import { AppBaseEntity } from '@backend/core/entity/appBaseEntity';
import { BaseId } from '../value-object/base_id';

/**
 * Enumの値が実装すべきメソッドを定義するインターフェース
 * PythonのEnumValueProtocolのTypeScript版
 */
export interface EnumValueProtocol<IdType extends BaseId, EntityType extends AppBaseEntity = AppBaseEntity> {
  /**
   * 値オブジェクトのIDを返す
   */
  readonly id: IdType;

  /**
   * エンティティから値を設定する
   * @param entity エンティティ
   */
  setFromEntity(entity: EntityType): void;
}
